cpp/src/Ice/SSL/SecureTransportTransceiverI.cpp

```diff
@@ -503,24 +503,22 @@ Ice::SSL::SecureTransport::TransceiverI::toDetailedString() const
 }
 
 Ice::ConnectionInfoPtr
-Ice::SSL::SecureTransport::TransceiverI::getInfo() const
+Ice::SSL::SecureTransport::TransceiverI::getInfo(bool incoming, string adapterName, string connectionId) const
 {
-    auto info = make_shared<Ice::SSL::ConnectionInfo>();
-    info->underlying = _delegate->getInfo();
-    info->incoming = _incoming;
-    info->adapterName = _adapterName;
+    assert(incoming == _incoming);
+    assert(adapterName == _adapterName);
+
+    SecCertificateRef peerCertificate = nullptr;
+
     if (_peerCertificate)
     {
-        SecCertificateRef peerCertificate = _peerCertificate.get();
+        peerCertificate = _peerCertificate.get();
         CFRetain(peerCertificate);
-        info->peerCertificate = peerCertificate;
-    }
-    else
-    {
-        info->peerCertificate = nullptr;
     }
 
-    return info;
+    return make_shared<Ice::SSL::ConnectionInfo>(
```

Comment:
A minor issue with this update, and other certificates. If the constructor throws, or `_delegate->getInfo()` throws. we leak the peerCertificate.

Reply:
It's not clear to me why the constructor or _delegate->getInfo would throw.